source/credentials_provider_sts.c

```diff
@@ -221,16 +221,27 @@ static int s_on_incoming_body_fn(struct aws_http_stream *stream, const struct aw
 </AssumeRoleResponse>
  */
 static bool s_on_node_encountered_fn(struct aws_xml_parser *parser, struct aws_xml_node *node, void *user_data) {
-    if (aws_byte_cursor_eq_ignore_case(&node->name, &s_assume_role_root_name) ||
-        aws_byte_cursor_eq_ignore_case(&node->name, &s_assume_role_result_name) ||
-        aws_byte_cursor_eq_ignore_case(&node->name, &s_assume_role_credentials_name)) {
+
+    struct aws_byte_cursor node_name;
+
+    if (aws_xml_node_get_name(node, &node_name)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "(id=%p): Could not get xml node name for s_on_node_encountered_fn",
```

Comment:
let's replace s_on_node_encountered_fn with something more descriptive, since there will eventually be a bunch of these